ir/reader.go

```diff
@@ -122,26 +132,72 @@ func (self *IrReader) ReadToFunc(r io.Reader, f func(ffi.LogEvent) bool) (ffi.Lo
 	}
 }
 
-// Read the IR stream using the io.Reader until [ffi.LogEvent.Timestamp] >=
-// time. Errors are propagated from ReadNextLogEvent.
-func (self *IrReader) ReadToEpochTime(r io.Reader, time ffi.EpochTimeMs) (ffi.LogEvent, error) {
-	return self.ReadToFunc(r, func(e ffi.LogEvent) bool { return e.Timestamp >= time })
+// Read the CLP IR stream until a [ffi.LogEventView] is greater than or equal to
+// the given timestamp. Errors are propagated from [ReadToFunc].
+func (self *Reader) ReadToEpochTime(
+	time ffi.EpochTimeMs,
+) (*ffi.LogEventView, error) {
+	return self.ReadToFunc(func(event *ffi.LogEventView) bool { return event.Timestamp >= time })
+}
+
+// Read the CLP IR stream until [strings/Contains] returns true for a
+// [ffi.LogEventView] and the given sub string. Errors are propagated from
+// [ReadToFunc].
+func (self *Reader) ReadToContains(substr string) (*ffi.LogEventView, error) {
+	fn := func(event *ffi.LogEventView) bool {
+		return strings.Contains(event.LogMessageView, substr)
+	}
+	return self.ReadToFunc(fn)
+}
+
+// Read the CLP IR stream until [strings/HasPrefix] returns true for a
+// [ffi.LogEventView] and the given prefix. Errors are propagated from
+// [ReadToFunc].
+func (self *Reader) ReadToPrefix(prefix string) (*ffi.LogEventView, error) {
+	fn := func(event *ffi.LogEventView) bool {
+		return strings.HasPrefix(event.LogMessageView, prefix)
+	}
+	return self.ReadToFunc(fn)
 }
```

Comment:
The implementations of these functions are functionally correct. But they can be more efficient if we create a wildcard query and execute a query search.

Reply:
This can be another PR.

Reply:
According to our discussion, the string methods can outperform wildcard searches since it could have better time complexity. We could rewrite these methods in the future PRs if needed.